### PR TITLE
Prevent side-effects if `error` is non-zero, without relying on PUSH.

### PR DESCRIPTION
**Pull request by apt1002**

### CODE REVIEW

File: src/extra.c [hunk 1]
```diff
@@ -366,8 +366,10 @@ static int extra_libc(smite_state *S)
             char *s2 = (char *)smite_native_address_of_range(S, str2, 0);
             if (s1 == NULL || s2 == NULL)
                 error = -9;
-            if (error == 0)
-                PUSH(rename(s2, s1));
+            if (error == 0) {
+                int res = rename(s2, s1);
```
Comment (rrthomas) [inline src/extra.c:370]: Given the guard `error == 0`, why is the intermediate variable `res` necessary?

File: src/extra.c [hunk 2]
```diff
@@ -377,8 +379,10 @@ static int extra_libc(smite_state *S)
             char *s = (char *)smite_native_address_of_range(S, str, 0);
             if (s == NULL)
                 error = -9;
-            if (error == 0)
-                PUSH(remove(s));
+            if (error == 0) {
+                int res = remove(s);
```
Comment (rrthomas) [inline src/extra.c:383]: As above.

File: src/extra.c (hunk 1 above)
Comment (apt1002) [reply src/extra.c:370]: It's not, you're right.

Comment (rrthomas) [reply src/extra.c:370]: Great, can you update the PR accordingly please?

**Comment by apt1002** [comment]
I retract this pull request in favour of a more sweeping reform (branch issue35).

File: src/extra.c (hunk 1 above)
Comment (apt1002) [reply src/extra.c:370]: I think this has already been obviated by my next pull request (branch issue35) which touched the same code.

**Pull request closed without merging**